spinoso-time/src/time/chrono/build.rs

```diff
@@ -50,6 +51,38 @@ impl Time {
             offset,
         }
     }
+
+    /// Creates a new `Time` object from the `seconds` and `sub_seconds_nanos`
+    /// since the Unix EPOCH with a local offset.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::Time;
+    /// let epoch = Time::at(0, 0);
+    /// let epoch_plus_1_nano = Time::at(0, 1);
+    /// ```
+    #[inline]
+    #[must_use]
+    pub fn at(seconds: i64, sub_second_nanos: i64) -> Self {
+        let offset = Offset::Local;
+
+        let timestamp = seconds + (sub_second_nanos / i64::from(NANOS_IN_SECOND));
+        let sub_second_nanos = sub_second_nanos % i64::from(NANOS_IN_SECOND);
+
+        // Handle negative sub_seconds_nanos
```

Comment:
In cases like these where the behavior is non obvious, I like to inline sample executions from an irb repl in the comments, similar to the ones here:

https://github.com/artichoke/artichoke/blob/1498e9d99d4313bfecfd98744cc4db162eb27ccf/artichoke-backend/src/convert/implicit.rs#L75-L88